model/modeldecoder/modeldecodertest/populator.go

```diff
@@ -0,0 +1,139 @@
+// Licensed to Elasticsearch B.V. under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Elasticsearch B.V. licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package modeldecodertest
+
+import (
+	"fmt"
+	"reflect"
+	"strings"
+
+	"github.com/elastic/apm-server/model/modeldecoder/nullable"
+	"github.com/elastic/beats/v7/libbeat/common"
+)
+
+// InitStructValues iterates through the struct fields represented by
+// the given reflect.Value and initializes all fields with
+// some arbitrary value.
+func InitStructValues(val reflect.Value) {
```

Comment:
I think it might lead to a bit of a cleaner API if you have the exported functions accept `interface{}` (which is expected to be a pointer to a struct), and call `reflect.ValueOf` on it internally.